### PR TITLE
changed logging to only support syslogng and elasticsearch

### PR DESCRIPTION
**Pull request by mzeevi**
When a user wants logging, now automatcially SyslogNGOutput and SyslogNGFlow objects are created. In addition, support for Splunk has been removed and only Elasticsearch Outputs are supported. A test case has been modified due to a known issue whiich should be fixed in the next release of logging-operator (https://github.com/kube-logging/logging-operator/issues/1140)

### CODE REVIEW

File: internal/kinds/capp/resource-managers/syslogngflow.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	if f.IsRequired(capp) {
+		generatedSyslogNGFlow := f.prepareResource(capp)
+		currentSyslogNGFlow := loggingv1beta1.SyslogNGFlow{}
+
+		resourceManager := rclient.ResourceBaseManagerClient{Ctx: f.Ctx, K8sclient: f.K8sclient, Log: f.Log}
+		err := f.K8sclient.Get(f.Ctx, types.NamespacedName{Namespace: capp.Namespace, Name: syslogNGFlowName}, &currentSyslogNGFlow)
+
+		if errors.IsNotFound(err) {
+			if err := resourceManager.CreateResource(&generatedSyslogNGFlow); err != nil {
+				f.EventRecorder.Event(&capp, corev1.EventTypeWarning, eventCappSyslogNGFlowCreationFailed, fmt.Sprintf("Failed to create SyslogNGFlow %s for Capp %s", syslogNGFlowName, capp.Name))
```
Comment (fran2810) [inline internal/kinds/capp/resource-managers/syslogngflow.go:99]: split the lines on `,` 

``` Go 

f.EventRecorder.Event(&capp, corev1.EventTypeWarning, eventCappSyslogNGFlowCreationFailed, 
fmt.Sprintf("Failed to create SyslogNGFlow %s for Capp %s", syslogNGFlowName, capp.Name))
```

**Review by fran2810 — COMMENTED** [review]
over all really good ! nice job few small things but i think maybe the resource mangers should be divided to actions because the the resource  managers the Createorupdate are to thick

File: internal/kinds/capp/resource-managers/resource_prepare_interface.go [hunk 2]
```diff
@@ -11,19 +11,23 @@ type ResourceManager interface {
...
+	eventCappEnabled  = "CappEnabled"
+
 	KnativeServing                        = "knativeServing"
-	Flow                                  = "flow"
-	Output                                = "output"
-	NFSPVC                                = "NfsPvc"
+	eventCappKnativeServiceCreationFailed = "KnativeServiceCreationFailed"
+
+	DomainMapping                        = "domainMapping"
+	eventCappDomainMappingCreationFailed = "DomainMappingCreationFailed"
```
[the remark this answers is not in the document]
Comment (mzeevi) [reply internal/kinds/capp/resource-managers/resource_prepare_interface.go:21]: I will handle this in the next PR which addresses resource managers

File: internal/kinds/capp/resource-managers/syslogngoutput.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+		if errors.IsNotFound(err) {
+			if err := resourceManager.CreateResource(&generatedSyslogNGOutput); err != nil {
+				logger.Error(err, "failed to create SyslogNGOutput")
+				o.EventRecorder.Event(&capp, corev1.EventTypeWarning, eventCappSyslogNGOutputCreationFailed, fmt.Sprintf("Failed to create SyslogNGOutput %s for Capp %s", syslogNGOutputName, capp.Name))
+				return err
+			}
+
+			logger.Info("Created SyslogNGOutput successfully")
+			o.EventRecorder.Event(&capp, corev1.EventTypeNormal, eventCappSyslogNGlSOutputCreated, fmt.Sprintf("Created SyslogNGOutput %s for Capp %s", syslogNGOutputName, capp.Name))
+		}
```
[the remark this answers is not in the document]
Comment (mzeevi) [reply internal/kinds/capp/resource-managers/syslogngoutput.go:141]: I'll maybe address the division in the next PR

**Review by fran2810 — APPROVED** [review]
/lgtm

**Pull request merged**